lib/compat/wordpress-6.3/class-gutenberg-rest-global-styles-controller-6-3.php

```diff
@@ -48,4 +89,76 @@ protected function prepare_links( $id ) {
 
 		return $links;
 	}
+
+	/**
+	 * Prepare a global styles config output for response.
+	 *
+	 * @since 5.9.0
+	 * @since 6.2 Handling of style.css was added to WP_Theme_JSON.
+	 * @since 6.3 Added support for site origin in global styles.
+	 *
+	 * @param WP_Post         $post Global Styles post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$raw_config                       = json_decode( $post->post_content, true );
+		$is_global_styles_user_theme_json = isset( $raw_config['isGlobalStylesUserThemeJSON'] ) && true === $raw_config['isGlobalStylesUserThemeJSON'];
+		$config                           = array();
+		if ( $is_global_styles_user_theme_json ) {
+			$origin = ( isset( $post->post_name ) && 'wp-global-styles-site' === $post->post_name ) ? 'site' : 'custom';
```

Comment:
Documenting what I learn: the only change for the `prepare_item_for_response` method is this line. Otherwise, it's a verbatim copy of the [existing 6.2 method](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.2/class-gutenberg-rest-global-styles-controller-6-2.php#L32).

Reply:
Yes, that's correct. That line and the following one (which will be using $origin value) are the only changes.